### PR TITLE
Add some warning in klist.h etc

### LINKED ISSUE #3591: Automatic switch to gamma_only calculation
### Background

For a larger system, when setting kspacing, only one k-point is often obtained. Actually, users can manually change `kspacing` to `gamma_only` to accelerate calculations.

### Describe the solution you'd like

If the program can determine whether to switch to `gamma_only` when there is only one k-point, that will save users from frequent changes to the INPUT. I am not sure if implementing this feature will result in significant code changes.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Notice Possible Changes of Behavior (Reminder only for developers)

_No response_

### Notice any changes of core modules (Reminder only for developers)

_No response_

### Notice Possible Changes of Core Modules (Reminder only for developers)

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

### LINKED ISSUE #4328: Berry phase calculation should be modified with precisely setting 'symmetry = -1'
### Details

Berry phase calculations and the corresponding docs should be modified with precisely setting `symmetry = -1`. I will submit a PR to fix.

### Have you read the online manual http://abacus.deepmodeling.com/en/latest/

- [X] Yes, I have read the online manual.

### Task list for Issue attackers (only for developers)

- [ ] Identify the specific section of the documentation with the issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified issue.
- [ ] Update the documentation to address the issue, following the suggested improvement.
- [ ] Ensure the updated documentation adheres to the project's documentation standards.
- [ ] Test the updated documentation to ensure it is clear and accurate.
- [ ] Review and incorporate any relevant feedback from users or developers.
- [ ] Publish the updated documentation and notify the issue reporter.

### PR DESCRIPTION
**Pull request by Luc1anoo**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #4328
Fix #3591

### Unit Tests and/or Case Tests for my changes
- A unit test is added for each new feature or bug fix.

### What's changed?
- Example: My changes might affect the performance of the application under certain conditions, and I have tested the impact on various scenarios...

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

### CODE REVIEW

**Comment by mohanchen** [comment]
Tests failed @Luc1anoo , need attention

**Comment by WHUweiqingzhou** [comment]
@Luc1anoo please link some issue.

File: source/module_esolver/esolver_ks_lcao_elec.cpp [hunk 1]
```diff
@@ -719,13 +719,20 @@ void ESolver_KS_LCAO<TK, TR>::nscf(void)
...
 
     // add by jingan
-    if (berryphase::berry_phase_flag && ModuleSymmetry::Symmetry::symm_flag != 1)
+    if (berryphase::berry_phase_flag)
     {
-        berryphase bp(this->LOC);
-        bp.lcao_init(
-            this->kv,
-            this->GridT); // additional step before calling macroscopic_polarization (why capitalize the function name?)
-        bp.Macroscopic_polarization(this->pw_wfc->npwk_max, this->psi, this->pw_rho, this->pw_wfc, this->kv);
```
Comment (WHUweiqingzhou) [inline source/module_esolver/esolver_ks_lcao_elec.cpp:728]: Why do you delete this line?

Comment (Luc1anoo) [reply source/module_esolver/esolver_ks_lcao_elec.cpp:728]: This is a mistake and I will fix it.

**Comment by WHUweiqingzhou** [comment]
I will pull a new PR, thus close this one now.

**Pull request closed without merging**